src/main/java/uk/gov/hmcts/reform/sscs/personalisation/Personalisation.java

```diff
@@ -229,19 +235,38 @@ protected Map<String, String> create(final SscsCaseDataWrapper responseWrapper,
             personalisation.put(JOINT_PARTY_APPEAL, "No");
         }
 
-        if (ccdResponse.getHearings() != null && !ccdResponse.getHearings().isEmpty()) {
-
-            Hearing latestHearing = NotificationUtils.getLatestHearing(ccdResponse);
-            if (latestHearing != null) {
-                HearingDetails latestHearingValue = latestHearing.getValue();
-                LocalDateTime hearingDateTime = latestHearingValue.getHearingDateTime();
-                personalisation.put(HEARING_DATE, formatLocalDate(hearingDateTime.toLocalDate()));
-                translateToWelshDate(hearingDateTime.toLocalDate(), ccdResponse, value -> personalisation.put(WELSH_HEARING_DATE, value));
-                personalisation.put(HEARING_TIME, formatLocalTime(hearingDateTime));
-                personalisation.put(VENUE_ADDRESS_LITERAL, formatAddress(latestHearing));
-                personalisation.put(VENUE_MAP_LINK_LITERAL, latestHearingValue.getVenue().getGoogleMapLink());
-                personalisation.put(DAYS_TO_HEARING_LITERAL, calculateDaysToHearingText(hearingDateTime.toLocalDate()));
+        Hearing latestHearing = ccdResponse.getLatestHearing();
+
+        HearingRoute hearingRoute = ccdResponse.getSchedulingAndListingFields().getHearingRoute();
+        if (HearingRoute.LIST_ASSIST == hearingRoute) {
+            latestHearing = ofNullable(ccdResponse.getHearings())
+                .orElse(Collections.emptyList()).stream()
+                .filter(hearing -> nonNull(hearing.getValue().getEpimsId()))
+                .max(Comparator.naturalOrder())
+                .orElse(null);
+        }
+
+        if (latestHearing != null) {
+            HearingDetails latestHearingValue = latestHearing.getValue();
+            LocalDateTime hearingDateTime = latestHearingValue.getHearingDateTime();
+            if (HearingRoute.LIST_ASSIST == hearingRoute) {
+                hearingDateTime = latestHearingValue.getStart();
             }
+
+            try {
+                mapper.registerModule(new JavaTimeModule());
```

Comment:
Can we move this out into a constructor?

Reply:
might be able to, this class doesn't already have a constructor I can add to and as it's so complex I didn't really want to mess around with it. I can try making a constructor if you want, might mean I have to redo all the related tests...

Reply:
Ah well, don't worry aboot it